app/src/main/java/jp/co/cyberagent/stf/MinitouchAgent.java

```diff
@@ -96,52 +108,73 @@ public static void main(String[] args) {
-        coords.y = (rotation * width)+(float)(x * Math.sin(-rad) + y * Math.cos(-rad));
-        MotionEvent event = MotionEvent.obtain(lastMouseDown, now, action, 1, pointerProperties,
-            pointerCoords, 0, buttonState, 1f, 1f, 0, 0,
+        double rad = Math.toRadians(rotation * 90.0);
+        coords.x = (float)(p.lastX * Math.cos(-rad) - p.lastY * Math.sin(-rad));
+        coords.y = (rotation * width)+(float)(p.lastX * Math.sin(-rad) + p.lastY * Math.cos(-rad));
+        return MotionEvent.obtain(p.lastMouseDown, now, p.action, idx+1, pointerProperties,
+            pointerCoords, 0, 0, 1f, 1f, 0, 0,
             InputDevice.SOURCE_TOUCHSCREEN, 0);
-        return event;
     }
 
-    public MinitouchAgent(int width, int height, Handler handler) {
-        this.width = width;
-        this.height = height;
-        this.handler = handler;
-        inputManager = new InputManagerWrapper();
-        windowManager = new WindowManagerWrapper();
-    }
+    private List<MotionEvent> getMotionEvent(PointerEvent p1, PointerEvent p2) {
+        List<MotionEvent> combinedEvents = new ArrayList<>(2);
+        long now = SystemClock.uptimeMillis();
+        if (p1.action != MotionEvent.ACTION_MOVE) {
+            combinedEvents.add(getMotionEvent(p1));
+            combinedEvents.add(getMotionEvent(p2,1));
+        } else {
+            MotionEvent.PointerCoords coords1 = pointerCoords[0];
+            MotionEvent.PointerCoords coords2 = pointerCoords[1];
+            int rotation = windowManager.getRotation();
+            double rad = Math.toRadians(rotation * 90.0);
 
-    @Override
-    public void run() {
-        try {
-            Log.i(TAG, String.format("creating socket %s", SOCKET));
-            serverSocket = new LocalServerSocket(SOCKET);
-        } catch (IOException e) {
-            e.printStackTrace();
-            return;
+            coords1.x = (float) (p1.lastX * Math.cos(-rad) - p1.lastY * Math.sin(-rad));
+            coords1.y = (rotation * width) + (float) (p1.lastX * Math.sin(-rad) + p1.lastY * Math.cos(-rad));
+
+            coords2.x = (float) (p2.lastX * Math.cos(-rad) - p2.lastY * Math.sin(-rad));
+            coords2.y = (rotation * width) + (float) (p2.lastX * Math.sin(-rad) + p2.lastY * Math.cos(-rad));
+
+            MotionEvent event = MotionEvent.obtain(p1.lastMouseDown, now, p1.action, 2, pointerProperties,
+                pointerCoords, 0, 0, 1f, 1f, 0, 0,
+                InputDevice.SOURCE_TOUCHSCREEN, 0);
+            combinedEvents.add(event);
         }
-        manageClientConnection();
-        try {
-            serverSocket.close();
+        return combinedEvents;
+    }
+
+    private void sendBanner(LocalSocket clientSocket) {
+        try{
+            OutputStreamWriter out = new OutputStreamWriter(clientSocket.getOutputStream());
+            out.write("v 1\n");
+            String resolution = String.format(Locale.getDefault(), "^ %d %d %d %d%n",
```

Comment:
Shouldn't we use Locale.US or ENGLISH here?
If locale is to e.g. `Locale.forLanguageTag("ar-SA-u-nu-arab")` then Easter-Arabic numeral will be sent to socket.

Reply:
You are right. I missed that.